### PR TITLE
Feature/chain fusion/include network tokens in manage modal

### PR DESCRIPTION
**Pull request by AntonioVentilii-DFINITY**
# Motivation

We include the list of all non IC tokens in component `ManageTokens`, respecting the visual order that the user sees in the single-view page (basically the ICRC tokens are last).
Furthermore we provide the ManageTokenToggle to the new tokens (for now disabled).

# Changes

- Created `allTokens` as merged list of `networkTokens` and `allIcrcTokens`. Note that we remove the duplicates, but IC tokens take priority. 
- Change index of loop EACH for `tokens`: it was giving duplicates indices, since the symbol may be repeated for different tokens on different networks (ICP native vs ICP Ethereum network).
- Use `ManageTokenToggle` for non-IC tokens.

# To-Do

- [ ] Remove ckLocal IC tokens from the list in case of Chain Fusion

# Tests

See below a local replica simulation:

Screen Recording 2024-06-19 at 13 16 20

### CODE REVIEW

**Comment by AntonioVentilii-DFINITY** [comment]
P.S. I noticed a delay in the overflow menu, that is being solved with PR #1497

File: src/frontend/src/icp-eth/components/tokens/ManageTokens.svelte [hunk 1]
```diff
@@ -51,6 +53,22 @@
 		)
 	].sort(sortIcTokens);
 
+	let allTokens: Token[] = [];
+	$: allTokens = [
```
Comment (peterpeterparker) [inline src/frontend/src/icp-eth/components/tokens/ManageTokens.svelte:57]: Can't we use the global store `$tokens` to get all the tokens?
Also shouldn't we filter test tokens?

File: src/frontend/src/icp-eth/components/tokens/ManageTokens.svelte [hunk 2]
```diff
@@ -60,12 +78,13 @@
...
-		? allIcrcTokens
-		: allIcrcTokens.filter(
-				({ name, symbol, alternativeName }) =>
-					name.toLowerCase().includes(filterTokens.toLowerCase()) ||
-					symbol.toLowerCase().includes(filterTokens.toLowerCase()) ||
-					(alternativeName ?? '').toLowerCase().includes(filterTokens.toLowerCase())
+		? allTokens
+		: allTokens.filter(
+				(token) =>
+					token.name.toLowerCase().includes(filterTokens.toLowerCase()) ||
```
Comment (peterpeterparker) [inline src/frontend/src/icp-eth/components/tokens/ManageTokens.svelte:84]: We compare token with their ids. Is there a specific reason why not to here?

File: src/frontend/src/icp-eth/components/tokens/ManageTokens.svelte [hunk 3]
```diff
@@ -141,7 +160,7 @@
 	</button>
 {:else}
 	<div class="container md:max-h-96 pr-2 pt-1 overflow-y-auto">
-		{#each tokens as token (token.symbol)}
+		{#each tokens as token (`${token.network.id.description}-${token.id.description}`)}
```
Comment (peterpeterparker) [inline src/frontend/src/icp-eth/components/tokens/ManageTokens.svelte:163]: Is this related to this PR?

File: src/frontend/src/icp-eth/components/tokens/ManageTokens.svelte (hunk 2 above)
Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/icp-eth/components/tokens/ManageTokens.svelte:84]: I actually kept the same searching logic: when the user search for a token, it checks the name, the symbol and, if there is, the alternative name

File: src/frontend/src/icp-eth/components/tokens/ManageTokens.svelte (hunk 3 above)
Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/icp-eth/components/tokens/ManageTokens.svelte]: Created PR #1498

File: src/frontend/src/icp-eth/components/tokens/ManageTokens.svelte (hunk 2 above)
Comment (peterpeterparker) [reply src/frontend/src/icp-eth/components/tokens/ManageTokens.svelte:84]: I see, thanks for the feedback.

File: src/frontend/src/icp-eth/components/tokens/ManageTokens.svelte (hunk 1 above)
Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/icp-eth/components/tokens/ManageTokens.svelte:57]: I used `networkTokens` since it will keep the same behavior if we select ICP network: if it's Chain Fusion, `$tokens` is the same as `$networkTokens` but without the test tokens (Sepolia for example).

However, you are right that the ICP test tokens appear there, I wanted to do a separate PR to remove them.

Do you prefer here?

Comment (peterpeterparker) [reply src/frontend/src/icp-eth/components/tokens/ManageTokens.svelte:57]: Is that the expected behavior from Artem and Mary? I actually would expect that when I access "Manage Tokens" I can always manage all my tokens regardless of the selected network. 🤷‍♂️

Comment (peterpeterparker) [reply src/frontend/src/icp-eth/components/tokens/ManageTokens.svelte:57]: Note that my comment above about testnet was incorrect I think. For ERC20 I can also set custom ERC20 on Sepolia.

Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/icp-eth/components/tokens/ManageTokens.svelte:57]: > Is that the expected behavior from Artem and Mary? I actually would expect that when I access "Manage Tokens" I can always manage all my tokens regardless of the selected network. 🤷‍♂️

Ah, that I didn't know, I thought the user should manage only the tokens in the selected network. Let me ask them

Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/icp-eth/components/tokens/ManageTokens.svelte:57]: I wrote to Artem and he said

Ok, it looks like it indeed m

> akes sense to show only the tokens related to the selected network. Is it possible?
> Maybe some small hint under the search would help with the context?
> e.g.
> Managing token for Internet Computer
> Managing token for ChainFusion

he made edits in figma

https://www.figma.com/design/2Wi7w1kenBrOhKgKo4iD3Z/OISY-Layouts?node-id=1354-5273&t=JO5QzcGQyyYI2EhX-0

Comment (peterpeterparker) [reply src/frontend/src/icp-eth/components/tokens/ManageTokens.svelte:57]: Thanks for double checking. I also quickly discussed with him to be sure I understand everything. Let's do that then.

**Review by peterpeterparker — APPROVED** [review]
LGTm, thx
